SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+		$result = GetVCServerPasswordPolicy -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName
+		$result.productAttributes | Add-Member -notepropertyname "driftAlarm" -notepropertyvalue ""
+		$result.productAttributes | Add-Member -notepropertyname "driftMessage" -notepropertyvalue ""
+		$ppmEnvironmentalDetails.Add($result)
+		Write-LogMessage -type INFO -Message "Retrieving VC server password policies..complete"
+		
+		##Retrieve SSO password policy details
+		$result = GetSingleSignOnPasswordPolicy -server $ppmVariables.ssoServerFqdn -user $ppmVariables.ssoServerUser -pass $ppmVariables.ssoServerPass
+		$result.productAttributes | Add-Member -notepropertyname "driftAlarm" -notepropertyvalue ""
+		$result.productAttributes | Add-Member -notepropertyname "driftMessage" -notepropertyvalue ""
+		$ppmEnvironmentalDetails.Add($result)
+		
+		##Retrieve NSX-T Manager password policy details
+		$nsxtManagerDetails = Get-NsxtServerDetail -fqdn $ppmVariables.sddcManagerFqdn -username $ppmVariables.sddcManagerUser -password $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName -listNodes
+		foreach ($nsxtManagerNode in $nsxtManagerDetails.nodes) {
+			$result = GetNsxtManagerPasswordPolicy  -server $nsxtManagerNode.fqdn -user $nsxtManagerDetails.adminUser -pass $nsxtManagerDetails.AdminPass
+			$result.productAttributes | Add-Member -notepropertyname "driftAlarm" -notepropertyvalue ""
+			$result.productAttributes | Add-Member -notepropertyname "driftMessage" -notepropertyvalue ""
+			$ppmEnvironmentalDetails.Add($result)
+		}		
+		##Retrieve NSX-T Edge password policy details
+		$results = GetNsxtEdgeNodePasswordPolicy -server $nsxtManagerDetails.fqdn -user $nsxtManagerDetails.adminUser -pass $nsxtManagerDetails.AdminPass
+		foreach ($result in $results) {
+			$result.productAttributes | Add-Member -notepropertyname "driftAlarm" -notepropertyvalue ""
+			$result.productAttributes | Add-Member -notepropertyname "driftMessage" -notepropertyvalue ""
+			$ppmEnvironmentalDetails.Add($result)
+		}
+		##Retrieve WSA Password policy details
+		if ($ppmVariables.wsaFqdn -and $ppmVariables.wsaAdminUser -and $ppmVariables.wsaAdminPass) {
+			Write-LogMessage -type INFO -Message "Retrieving WSA server password policies.."
+			$result = GetWSAPasswordPolicyAll -server $ppmVariables.wsaFqdn -user $ppmVariables.wsaAdminUser -pass $ppmVariables.wsaAdminPass
+			$result.productAttributes | Add-Member -notepropertyname "driftAlarm" -notepropertyvalue ""
+			$result.productAttributes | Add-Member -notepropertyname "driftMessage" -notepropertyvalue ""
+			$ppmEnvironmentalDetails.Add($result)
+			Write-LogMessage -type INFO -Message "Retrieving WSA server password policies..complete"
+		} else {
+			Write-LogMessage -Type WARNING -Message "Workspace One Access Skipped.  Login details are not provided."  
```

Comment:
"Workspace ONE Access"

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e